salt/daemons/masterapi.py

```diff
@@ -717,7 +717,7 @@ def _pillar(self, load):
                              'data',
                              {'grains': load['grains'], 'pillar': data})
             if self.opts.get('minion_data_cache_events') is True:
-                self.event.fire_event('Minion data cache refresh', salt.utils.event.tagify(load['id'], 'refresh', 'minion'))
+                self.event.fire_event({'comment': 'Minion data cache refresh'}, salt.utils.event.tagify(load['id'], 'refresh', 'minion'))
```

Comment:
What is the reason for switching the data from a string to a dict?

Reply:
This one is to make the events consistent across multiple transports

Reply:
As I see `salt.utils.event.SaltEvent.fire_event()` raises an error if data is something except a `dict`.